vissl/data/ssl_transforms/tensor_to_img_pil.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+from io import BytesIO
+
+from classy_vision.dataset.transforms import register_transform
+from classy_vision.dataset.transforms.classy_transform import ClassyTransform
+from PIL import Image
+
+
+@register_transform("TensorToImgPil")
+class TensorToImgPil(ClassyTransform):
+    """
+    Tensor to Pil Image. Currently Used for OnBoxDataLoader, as images are
+    returned as tensors, but other transforms expect an Image PIL.
+    """
+
+    def __call__(self, sample):
+        # Convert tensors to image pil for subsequent transforms.
+        for i, tensor in enumerate(sample["data"]):
+            sample["data"][i] = self._tensor_to_img_pil(tensor)
+
+        sample = self._reformat_sample(sample)
+
+        return sample
+
+    def _tensor_to_img_pil(self, tensor):
```

Comment:
There is a torchvision transformation that might be used here (to move from a tensor to a PIL image): https://pytorch.org/vision/stable/_modules/torchvision/transforms/transforms.html#ToPILImage

Reply:
So I tried this, it doesn't work unfortunately. The tensor here is flattened, and torchvision expects a C x W x H tensor.